pkg/process/metadata/parser/java/yaml.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"io"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type flattener struct {
+	props map[string]string
+}
+
+func (f *flattener) visit(path []string, node *yaml.Node) {
+	for i := 0; i < len(node.Content); i += 2 { // should be safer
+		current := node.Content[i]
+		next := node.Content[i+1]
+		if current.Kind != yaml.ScalarNode || !(next.Kind == yaml.ScalarNode || next.Kind == yaml.MappingNode) {
```

Comment:
I'm not sure I understand the line here.

Reply:
It is interested to parse only the Node if the key (current) is convertible to a string and the value either is a scalar value or a is a nested object (it will hence recurse for the latter).

It is not general purpose code to flatten whatever yaml but mainly tailored to extract single values like `spring.application.name`.

Reply:
updated